plugins/mac-virtualcam/src/obs-plugin/plugin-main.mm

```diff
@@ -316,7 +316,7 @@ static void virtualcam_output_raw_video(void *data, struct video_data *frame)
 }
 
 struct obs_output_info virtualcam_output_info = {
-	.id = "virtualcam_output",
+	.id = "dal_output",
```

Comment:
> Use a different ID for CMIO, period. It makes no sense to use the same for different implementations.

Yes, it does. We are not going to have two virtual camera implementations on macOS running at the same time, as there are no benefits to have the old DAL plugin (which only works with a couple apps) over the CMIO extension (which works with literally every app that supports a camera). On the OBS side, more than half of the existing code is re-used to send the frames to the extension. It will be the same OBS output.

And quite frankly, I'm very annoyed by the your comment. People can have different opinions on things, and opinions can differ (like my *opinion* on changing this in the first place). Saying "You have to do X, period" is anything but helpful (to the point where I honestly don't have any interest in continuing this conversation), especially if your knowledge of the situation is limited.

Reply:
I didn't mean to be rude, but it's been more than 1 year that this goes in the direction of different ID for every implementation.

Okay, macOS does not a transition period because the transition will be invisible.

Linux needs one, we actually rely on V4L2 (kernel space) and we need to transition to PipeWire (user space) which is not available in all (non out-of-support) distros.

ID should be unique per implementation, it's like giving the same ID to DeckLink and AJA output because they have the same purpose even if they rely on the different hardware.

And being per OS is not a reason to do it either because when we reach the point where a new implementation arise we need a new ID.